pennylane/qchem/convert.py

```diff
@@ -354,3 +355,264 @@ def import_operator(qubit_observable, format="openfermion", wires=None, tol=1e01
+        \mathrm{unocc}} \hat{c}_r^\dagger \hat{c}_s^\dagger \hat{c}_p \hat{c}_q.
+
+
+    In the equations above the indices :math:`p, q, r, s` run over the
+    occupied (occ) and unoccupied (unocc) *spin* orbitals and :math:`\hat c` and
+    :math:`\hat c^\dagger` are the electron annihilation and creation operators,
+    respectively.
+
+    Args:
+        electrons (int): number of electrons
+        orbitals (int): number of spin orbitals
+
+    Returns:
+        tuple(list, list): lists with the indices of the spin orbitals involved in the excitations
+
+    **Example**
+
+    >>> electrons = 2
+    >>> orbitals = 4
+    >>> _excitations(electrons, orbitals)
+    ([[[0, 2]], [[0, 3]], [[1, 2]], [[1, 3]]], [[0, 1, 2, 3]])
+    """
+    singles_p, singles_q = [], []
+    doubles_pq, doubles_rs = [], []
+
+    for i in range(electrons):
+        singles_p += [i]
+        doubles_pq += [[k, i] for k in range(i)]
+    for j in range(electrons, orbitals):
+        singles_q += [j]
+        doubles_rs += [[k, j] for k in range(electrons, j)]
+
+    singles = [[p] + [q] for p in singles_p for q in singles_q]
+    doubles = [pq + rs for pq in doubles_pq for rs in doubles_rs]
+
+    return singles, doubles
+
+
+def _excited_configurations(electrons, orbitals, excitation):
+    r"""Generate excited configurations from a Hartree-Fock reference state.
+
+    This function generates excited configurations in the form of integers representing a binary
+    string, e.g., :math:`|1 1 0 1 0 0 \rangle is represented by :math:`int('110100', 2) = 52`.
+
+    The excited configurations are generated from a Hartree-Fock (HF) reference state. The HF state
+    is assumed to have the form :math:`|1 1 ...1 0 ... 0 0 \rangle` where the number of :math:`1`
+    and :math:`0` elements are the number of occupied and unoccupied spin orbitals, respectively.
+    The string representation of the state is obtained by converting the occupation-number vector to
+    a string, e.g., ``111000`` to represent :math:`|1 1 1 0 0 0 \rangle.
+
+    Each excited configuration has a sign, :math:`+1` or :math:`-1`, that is obtained by reordering
+    the creation operators.
+
+    Args:
+        electrons (int): number of electrons
+        orbitals (int): number of spin orbitals
+        excitation (int): number of excited electrons
+
+    Returns:
+        tuple(array, array): arrays of excited configurations and signs obtained by reordering the
+         creation operators
+
+    **Example**
+
+    >>> electrons = 2
+    >>> orbitals = 5
+    >>> excitation = 2
+    >>> _excitated_states(electrons, orbitals, excitation)
+    (array([28, 26, 25]), array([ 1, -1,  1]))
+    """
+    hf_state = qml.qchem.hf_state(electrons, orbitals)
+    singles, doubles = _excitations(electrons, orbitals)
+    states, signs = [], []
+
+    if excitation == 1:
+        for s in singles:
+            state = hf_state.copy()
+            state[s] = state[s[::-1]]
```

Comment:
so `s` is a doubly nested list, and here we are reversing the order. Is this like computing a Transpose or something?

Reply:
It is just a simple permutation, something like `state[[0, 2]] =  state[[2, 0]]` for exciting one electron from the 0-th orbital to the 2-nd orbital.